src/jit/MemoryInl.h

```diff
@@ -215,11 +246,61 @@ void MemAddress::load(sljit_compiler* compiler)
 #endif /* HAS_SIMD */
 }
 
+#if defined(ENABLE_EXTENDED_FEATURES) && (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+static void atomicRmwLoad64(std::atomic<int64_t>& shared, uint64_t* result)
```

Comment:
Is this valid to pas a pointer as `std::atomic<int64_t>&`? For me it would be safer to pass the pointer, and convert it to this special type.